clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/validators/EcsCreateServerGroupDescriptionValidator.java

```diff
@@ -155,4 +159,39 @@ public void validate(List priorDescriptions, Object description, Errors errors)
       }
     }
   }
+
+  private void validateTargetGroupMappings(
+      CreateServerGroupDescription createServerGroupDescription, Errors errors) {
+    if (createServerGroupDescription.getTargetGroupsMappings() != null) {
+      Set<CreateServerGroupDescription.TargetGroupProperties> targetGroupMappings =
+          createServerGroupDescription.getTargetGroupsMappings();
+
+      for (CreateServerGroupDescription.TargetGroupProperties targetGroupProperties :
+          targetGroupMappings) {
+        // Verify each target group mapping contains a target group name, container name, and
```

Comment:
re: requiring "container name" - is it the case at this point that a customer using inputs (with one container) would have entered multiple target group mappings w/o a 'container name'? B/C in that situation the name of the container is not defined until we're executing the server group deployment. 

We can check if this (1 container defined w/ inputs) is the case by seeing whether 'dockerImageAddress' is populated.

Reply:
I don't like the idea of hard coding in magic numbers if we can avoid it. Is there a property that's only defined if it's using inputs vs artifacts?

Reply:
Sounds good!